sandman/source/common/non_null.h

```diff
@@ -13,7 +16,7 @@ template <typename ReturnT, typename... ParametersT>
 class Common::NonNull<ReturnT (*)(ParametersT...)>
```

Comment:
Should I remove this class? It is used in the input buffer to not have to explicitly check that a function pointer is null.

Reply:
Yeah, but in a separate pull request. I think having function pointers in particular that are null in this code base are pretty easily avoidable mistakes, but it does strike me that it would be good to have some sort of assert mechanism or maybe we can detect the presence of a null function at compile time?